src/csharp/Microsoft.Spark/Sql/Types/Date.cs

```diff
@@ -0,0 +1,94 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+
+namespace Microsoft.Spark.Sql.Types
+{
+    /// <summary>
+    /// Represents Date containing year, month, and day.
+    /// </summary>
+    public class Date
+    {
+        /// <summary>
+        /// Constructor for Date class.
+        /// </summary>
+        /// <param name="dt">DateTime object</param>
+        public Date(DateTime dt)
+        {
+            Year = dt.Year;
+            Month = dt.Month;
+            Day = dt.Day;
+        }
+
+        /// <summary>
+        /// Constructor for Date class.
+        /// </summary>
+        /// <param name="year">The year (1 through 9999)</param>
+        /// <param name="month">The month (1 through 12)</param>
+        /// <param name="day">The day (1 through the number of days in month)</param>
+        public Date(int year, int month, int day)
+        {
+            Year = year;
```

Comment:
Can you check `DateTime` implementation if they check against the range?

Reply:
https://referencesource.microsoft.com/#mscorlib/system/datetime.cs

Reply:
Yes, it will check the range.